src/grappler/ngraph_optimizer.cc

```diff
@@ -47,7 +47,7 @@ Status NgraphOptimizer::Init(
     NGRAPH_VLOG(3) << config_backend_name;
     std::vector<std::string> additional_attributes =
         BackendManager::GetBackendAdditionalAttributes(config_backend_name);
-    for (int i = 0; i < additional_attributes.size(); i++) {
+    for (unsigned int i = 0; i < additional_attributes.size(); i++) {
```

Comment:
Change this to `auto` or `size_t`. Also do not use c style cast- always use C++ case i.e., siatic_cast<> etc. when needed.